injection/sharedmain/main.go

```diff
@@ -372,14 +378,53 @@ func SetupLoggerOrDie(ctx context.Context, component string) (*zap.SugaredLogger
 
 // SetupObservabilityOrDie sets up the observability using the config from the given context
 // or dies by calling log.Fatalf.
-func SetupObservabilityOrDie(ctx context.Context, component string, logger *zap.SugaredLogger, profilingHandler *profiling.Handler) {
-	observabilityConfig, err := GetObservabilityConfig(ctx)
+func SetupObservabilityOrDie(
+	ctx context.Context,
+	component string,
+	logger *zap.SugaredLogger,
+	pprof *pprofServer,
+) (*metrics.MeterProvider, *tracing.TracerProvider) {
+	cfg, err := GetObservabilityConfig(ctx)
 	if err != nil {
 		logger.Fatal("Error loading observability configuration: ", err)
 	}
-	observabilityConfigMap := observabilityConfig.GetConfigMap()
-	metrics.ConfigMapWatcher(ctx, component, SecretFetcher(ctx), logger)(&observabilityConfigMap)
-	profilingHandler.UpdateFromConfigMap(&observabilityConfigMap)
+
+	pprof.UpdateFromConfig(cfg.Runtime)
+
+	resource := getResource(component)
```

Comment:
Do we want `getResource` to be in `observability` for sharing with non-`sharedmain` binaries?